Source/ASCollectionView.mm

```diff
@@ -2311,7 +2311,11 @@ - (void)rangeController:(ASRangeController *)rangeController updateWithChangeSet
       
       // Flush any range changes that happened as part of submitting the update.
       as_activity_scope(changeSet.rootActivity);
-      [self->_rangeController updateIfNeeded];
+      if (numberOfUpdates > 0 && ASActivateExperimentalFeature(ASExperimentalRangeUpdateOnChangesetUpdate)) {
+        [self->_rangeController setNeedsUpdate];
```

Comment:
When we first talked about this diff, I was thinking about calling `setNeedsUpdate` in the `completion` block ([here](https://github.com/TextureGroup/Texture/pull/1976/files#diff-a763091124ee49f35c39703ce93f1d38fb204a058ff6483896416ffa678aa155R2258)). But thinking about it more, I think you're right that it's better do so here because the any range changes is the result of submitting the update (instead of the following update animations).

Reply:
Ah I was actually unaware of this subtlety. This makes sense though